README.md

```diff
@@ -108,16 +108,44 @@ skew may result in `system time has changed` messages in VM logs.
 #### Network
 
 The guest agent uses network interface metadata to manage the network
-interfaces in the guest by performing the following tasks:
-
-*   Enabled all associated network interfaces on boot.
+interfaces by performing the following tasks:
+
+*   Enable all associated network interfaces on boot.
+    *   Detect the current active network manager service that is managing the
+        primary NIC.
+    *   Rollback and delete any files/configurations left behind by all other
+        supported network manager services.
+    *   Write and apply new configurations for the secondary NICs using the
+        network manager service detected in the first step.
 *   Setup or remove IP routes in the guest for IP forwarding and IP aliases
     *   Only IPv4 IP addresses are currently supported.
     *   Routes are set on the primary ethernet interface.
     *   Google routes are configured, by default, with the routing protocol ID
         `66`. This ID is a namespace for daemon configured IP addresses. It can
         be changed with the config file, see below.
 
+On Linux, supported network managers are (listed by priority):
+
+*   `netplan`
+*   `wicked`
+*   `NetworkManager`
+*   `systemd-networkd`
+*   `dhclient`
+
+If none of the first 4 network manager services are detected on the system, then
+the agent will default to using `dhclient` for managing network interfaces.
+
+The following configuration flags can control the behavior:
+
+*   `manage_primary_nic`: When enabled, the agent will start managing the
+    primary NIC in addition to the secondary NICs.
+
+For more information about the instance configuration, see the Configuration
+section.
+
+The guest agent will also setup VLANs if VLAN is enabled. The setup and
```

Comment:
can you also add a note here about agent applying config for vlan nic whose parent is primary nic regardless of `manage_primary_nic` entry

Reply:
Done.